src/attributes/codegen.md

```diff
@@ -133,7 +134,8 @@ Feature     | Implicitly Enables | Description
 
 #### `wasm32` or `wasm64`
 
-This platform allows `#[target_feature]` to be applied to both safe and
+Executing code with unsupported features is allowed (i.e., is not UB) on this platform.
```

Comment:
```suggestion
Attempting to use instructions unsupported by the wasm engine will fail at load time
without the risk of being interpreted in a way different from what the compiler expected.
```

Reply:
I think it should explicitly say something about there being no Rust surface UB -- as the UB list says, this is defined per-target, so the targets should explicitly define it.

basically, there's a `bool` per target whether this is UB or not; we need to say explicitly which value taht `bool` ahs for wasm.

Reply:
I like the way bjorn3 has phrased it here. If you specifically want to make sure that the term "undefined behavior" shows up here, then I think a small adjustment could be made:

> `#[target_feature]` may be used with both safe and unsafe functions on Wasm platforms.
> Although it is undefined behavior to execute unsupported instructions, attempting to use instructions unsupported by the Wasm engine will fail at load time without the risk of being interpreted in a way different from what the compiler expected.

Reply:
> Although it is undefined behavior to execute unsupported instructions

The thing is though that this is *not* UB. Rust makes it safe to invoke a function with the wrong target_feature on wasm, so if that was UB that would be unsound.

(The UB we care about here is Rust UB. It is the Rust backend responsibility to worry about target UB, and certainly of no concern for the Rust Reference to worry about what is or is not UB in wasm programs.)

Reply:
I understand there are two perspectives of UB, but that is a very fine distinction that doesn't come across in the current wording. The statement that "Executing code with unsupported features is allowed on this platform" by itself doesn't come across as being correct to me because the platform does not allow executing code with unsupported features (sort of the exact opposite of what is written). I understand that you're trying to say Rust doesn't care, but it isn't coming across that way to me.

Perhaps another crack at it:

> `#[target_feature]` may be used with both safe and unsafe functions on Wasm platforms.
> It is not undefined behavior to use the `#[target_feature]` attribute because attempting to use instructions unsupported by the Wasm engine will fail at load time without the risk of being interpreted in a way different from what the compiler expected.

Reply:
Yes I like that, thanks! I just slightly changed the second sentence.